src/Umbraco.Web.UI.Client/src/views/propertyeditors/mediapicker/mediapicker.controller.js

```diff
@@ -4,23 +4,26 @@ angular.module('umbraco').controller("Umbraco.PropertyEditors.MediaPickerControl
     function ($rootScope, $scope, dialogService, entityResource, mediaResource, mediaHelper, $timeout, userService, $location, localizationService) {
 
         //check the pre-values for multi-picker
-        var multiPicker = $scope.model.config.multiPicker && $scope.model.config.multiPicker !== '0' ? true : false;
+        var multiPicker = $scope.model.config.maxNumber && parseInt($scope.model.config.maxNumber) > 1 ? true : false;
         var onlyImages = $scope.model.config.onlyImages && $scope.model.config.onlyImages !== '0' ? true : false;
         var disableFolderSelect = $scope.model.config.disableFolderSelect && $scope.model.config.disableFolderSelect !== '0' ? true : false;
 
         if (!$scope.model.config.startNodeId) {
-            userService.getCurrentUser().then(function(userData) {
+            userService.getCurrentUser().then(function (userData) {
                 $scope.model.config.startNodeId = userData.startMediaIds.length !== 1 ? -1 : userData.startMediaIds[0];
                 $scope.model.config.startNodeIsVirtual = userData.startMediaIds.length !== 1;
             });
         }
 
+
         function setupViewModel() {
             $scope.images = [];
             $scope.ids = [];
 
             $scope.isMultiPicker = multiPicker;
 
+            $scope.maxNumber = parseInt($scope.model.config.maxNumber);
+            $scope.minNumber = parseInt($scope.model.config.minNumber);

```

Comment:
I think this will fail if `maxNumber` or `minNumber` is null/undefined. You could fallback to `0` and handle it as no limitation or fallback to `null` and handle this where the scope properties are used.

```
$scope.maxNumber = $scope.model.config.maxNumber ? parseInt($scope.model.config.maxNumber) : 0;
$scope.minNumber = $scope.model.config.maxNumber ? parseInt($scope.model.config.minNumber) : 0;
```

```
$scope.maxNumber = $scope.model.config.maxNumber ? parseInt($scope.model.config.maxNumber) : null;
$scope.minNumber = $scope.model.config.maxNumber ? parseInt($scope.model.config.minNumber) : null;
```

Reply:
I have checked those line and if 0 it returns NaN. It doesn't seem to fail what your solution makes it more robust.